src/Symfony/Bundle/FrameworkBundle/Command/ContainerDebugCommand.php

```diff
@@ -166,6 +166,21 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         try {
             $helper->describe($io, $object, $options);
 
+            if (isset($options['id'])) {
+                if ($object->hasDefinition($options['id'])) {
+                    $definition = $object->getDefinition($options['id']);
+                    if ($definition->isDeprecated()) {
+                        $errorIo->warning($definition->getDeprecation($options['id'])['message'] ?? sprintf('The "%s" service is deprecated.', $options['id']));
```

Comment:
this should only be done in text format IMO. Structured formats (markdown, json, xml) should include that as a structured info instead IMO (allowing the consumer of that command to get the info, as json or xml are commonly used for scripting purposes).

Reply:
@stof thx for the comment

in fact it was [the first implementation I've done](https://github.com/symfony/symfony/pull/46814#issuecomment-1173506540), added this deprecated text just after the `autowire` output of any formats

So, if we are now on a [consensus as requested upper ](https://github.com/symfony/symfony/pull/46814#issuecomment-1173146383) I can:

text default format: keep it like this (so just display this warning in case of output txt)
for all others (json/xml/md): re-add what have been done earlier

is it ok? thank you

Reply:
friendly ping @ogizanagi @GromNaN as early reviewers, your thoughts on this?

Reply:
I agree that the info seems relevant as part of the standard output for formats other than text.